### PR TITLE
Organized lane marker UI classes

### PR DESCRIPTION
**Pull request by kianzarrin**
as preparation for #681 and #682 I organized lane marker UI classes a bit. this review is mostly renaming stuff and a bit of encapsulation. the code path is the same.

I moved `SegmentLaneMarker` and `NodeLaneMarker` into separate files turning them into independent classes.
I introduced a private LaneEnd master class that is responsible for Sheath overlay in lane connector. The `LaneEnd` class has stolen most of the functionality of the `NodeLaneMarker`.

EDIT: test result:
Screenshot (488)

About that `HOT REALOD` label in the image above, it really means in game hot reload:
- if you see HOT RELOAD it means that there has been hot reload (in game)
- if you don't see HOT RELOAD that does not mean there was no hot-reload ... there could be a hot reload in the main menu.

I hope this does not mislead anyone in the feature. maybe I should rename it to IGHR(in game hot reload).

### CODE REVIEW

**Comment by aubergine10** [comment]
Regarding splitting out the overlays stuff, see also: #630

For the in-game hot reload, I'd suggest ditching the mod name prefix (as a dev will already know exactly what they just built) so the toolbar title would be just `In-game hot-swap {Version.Revision}` - the revision just gives some indication of change if there are multiple hotloads.

File: TLM/TLM/Constants.cs [hunk 1]
```diff
@@ -40,5 +40,8 @@ public static class Constants {
         }
 
         public static IManagerFactory ManagerFactory => Manager.Impl.ManagerFactory.Instance;
+
+        public  const float MAX_HIT_ERROR = 2.5f;
```
Comment (kvakvs) [inline TLM/TLM/Constants.cs]: Add a comment please what is this (maybe also where this is to be used) and the unit (metres, probably)

**Comment by kianzarrin** [comment]
> Consider making a subdirectory there somewhere and move files to it, don't mix highlight modules with tool modules. You don't have to, but maybe makes sense?

Its already in the helpers directory.

**Comment by kianzarrin** [comment]
Is this ready to merge? multiple other tasks rely on this.

**Pull request merged**